### DISCUSSION TITLE
Next intl has an error in Next.js 14

### DISCUSSION THREAD

**Discussion by Half-Sugar-Code** [amannn/next-intl#675] [Q&A]
### Description

I have a question, the dependent version is Next.js 14.

I am based on https://i18nexus.com/tutorials/nextjs/next-intl Using a case study to edit the code, I found the following errors.

terminal： 
error：node_modules\next-intl\dist\development\config.js (6:8) @ getConfig

console
error1: Couldn't find next-intl config file. Please follow the instructions at https://next-intl-docs.vercel.app/docs/getting-started/app-router-server-components

error2: http://localhost:3000/de 404 (Not Found)
error3： validateDOMNesting(...): <div> cannot appear as a child of <#document>.
error4：DOMException: Failed to execute 'appendChild' on 'Node': Only one element on document allowed.

I see a link: https://next-intl-docs.vercel.app/docs/getting-started/app-router-server-components

But I can't open it  -.-

### Mandatory reproduction URL (CodeSandbox or GitHub repository)

https://github.com/SuperCodeCandy/Intl-error-demo/tree/main

### Reproduction description

Can be viewed https://github.com/SuperCodeCandy/Intl-error-demo/tree/main

### Expected behaviour

I'm not sure if it's because next-intl doesn't support Next.js14.

**Comment by amannn** [comment]
Hi, the tutorial you're referencing seems to be slightly outdated. Please follow [the official getting started guide](https://next-intl-docs.vercel.app/docs/getting-started/app-router) or alternatively check out [the App Router example](https://next-intl-docs.vercel.app/examples).

I'll move this to a discussion since it's not a bug report but a usage question.

**Reply by Half-Sugar-Code** [reply]
Oh, first of all, thank you very much for your reply. My demo uses https://i18nexus.com/tutorials/nextjs/next-intl I will try using the tutorial on the new address you provided for the case provided above. Thank you.

**Reply by Half-Sugar-Code** [reply]
Hi!
When attempting to connect using the learning document you provided me, I apologize for the following error on the console. I was unable to find a solution to the previous issues.

IntlError: MISSING_MESSAGE: No messages were configured on the provider.
    at getMessagesOrError (webpack-internal:///(rsc)/./node_modules/use-intl/dist/development/createFormatter-4848f6c8.js:100:27)
    at Object.createBaseTranslator (webpack-internal:///(rsc)/./node_modules/use-intl/dist/development/createFormatter-4848f6c8.js:118:29)
    at createTranslatorImpl (webpack-internal:///(rsc)/./node_modules/use-intl/dist/development/core.js:15:28)     
    at createTranslator (webpack-internal:///(rsc)/./node_modules/use-intl/dist/development/core.js:35:12)
    at eval (webpack-internal:///(rsc)/./node_modules/next-intl/dist/esm/react-server/getBaseTranslator.js:15:75) {  code: 'MISSING_MESSAGE',
  originalMessage: 'No messages were configured on the provider.'
}

**Reply by Half-Sugar-Code** [reply]
I have identified the cause of the error message. Thank you

**Reply by murbanowicz** [reply]
what was the reason?

**Comment by viclafouch** [comment]
Hey ! I've just got the same issue and I find the cause issue... I have the exact same config as the repo "App demo"

**Reply by viclafouch** [reply]
Just by doing this 

```tsx
import React from 'react'
import { useTranslations } from 'next-intl'

const Page = () => {
  const { t } = useTranslations()

  return <div>Hello</div>
}

export default Page

```

I got 

` validateDOMNesting(...): <div> cannot appear as a child of <#document>.`

What is happening ???

**Reply by moritzjacobs** [reply]
Not sure if still relevant, but you're missing "use client".

**Comment by zerosoul** [comment]
I have same problems

**Reply by Half-Sugar-Code** [reply]
Hi, The tutorial on that website is outdated. You can try following the tutorial on this [website](https://next-intl-docs.vercel.app/docs/getting-started/app-router)

**Comment by savchukartur** [comment]
I had the same error. Thanks to the [example](https://github.com/amannn/next-intl/tree/main/examples/example-app-router), I found that in the middleware.ts file, in the exported config.matcher, you need to add the following line: `'/((?!_next|_vercel|.\..).*)'`

After doing this, the error disappeared

**Comment by mahdipakravan-dev** [comment]
I had this error and resolved by these steps :

1. ensure all of the page have setRequestLocal(locale) function
2. add these 2 line to your main layout

`export function generateStaticParams() {
  return routing.locales.map((locale) => ({ locale }))
}

export async function generateMetadata({ params: { locale } }: { params: any }) {
  const t = await getTranslations({ locale, namespace: 'LocaleLayout' })

  return {
    title: t('title'),
  }
}`

3.ensure this code is available in your middleware config 
` '/((?!_next|_vercel|.\..).*)'`
